featuregate/registry.go

```diff
@@ -5,12 +5,22 @@ package featuregate // import "go.opentelemetry.io/collector/featuregate"
 
 import (
 	"fmt"
+	"net/url"
+	"regexp"
 	"sort"
 	"sync"
 	"sync/atomic"
+
+	"github.com/hashicorp/go-version"
 )
 
-var globalRegistry = NewRegistry()
+var (
+	globalRegistry = NewRegistry()
+
+	// idRegexp is used to validate the ID of a Gate.
+	// IDs' characters must be alphanumeric or dots.
+	idRegexp = regexp.MustCompile(`^[0-9a-zA-Z\.]*$`)
```

Comment:
Do we not allow "-" or "_"? Just a question, this may fail some already created featuregates. Happy to not allow that initially if we at least check contrib.

Reply:
> Do we not allow "-" or "_"? Just a question, this may fail some already created featuregates.

I think `-` can be confusing, at least on the first value (since we use `-` on the CLI to say when we disable a gate). We don't have feature gates with `-` or `_` on contrib (I manually checked and also would expect contrib tests to fail otherwise).

> Happy to not allow that initially if we at least check contrib.

I would lean towards being conservative here at first if that's okay with everybody else